src/java.base/share/classes/java/util/stream/Stream.java

```diff
@@ -1162,6 +1162,36 @@ default DoubleStream mapMultiToDouble(BiConsumer<? super T, ? super DoubleConsum
      */
     <R, A> R collect(Collector<? super T, A, R> collector);
 
+    /**
+     * Accumulates the elements of this stream into a {@code List}. The elements in
+     * the list will be in this stream's encounter order, if one exists. There are no
+     * guarantees on the implementation type, mutability, serializability, or
```

Comment:
It would be useful for callers to feel more confident that they will get an immutable instance. In java.time.* we have wording like "This interface places no restrictions on the mutability of implementations, however immutability is strongly recommended." Could something like that work here, emphasising that everyone implementing this method should seek to return an immutable list?

Reply:
Yes, good point, the "no guarantee of mutability" clashes with the later statement about the possibility of the returned instance being value-based, which strongly implies immutability. I'll work on tuning this up to be a stronger statement on immutability, while retaining "no-guarantee" for implementation type, serializability, etc. I think we do want to preserve future implementation flexibility in those areas.